sway-lsp/src/server_state.rs

```diff
@@ -114,6 +126,8 @@ pub struct CompilationContext {
     pub sync: Option<Arc<SyncWorkspace>>,
     pub token_map: Arc<TokenMap>,
     pub engines: Arc<RwLock<Engines>>,
+    pub compiled_programs: Option<Arc<CompiledPrograms>>,
+    pub runnables: Option<Arc<RunnableMap>>,
```

Comment:
We seem to `unwrap` these at call sites; id reccomend these not to be optional types in this case.
I do see that we only use the `None` variant in tests; in this case the tests should use concrete types aswell - or you can do something like so:

```suggestion
    #[cfg(not(test))]
    pub compiled_programs: Arc<CompiledPrograms>,
    #[cfg(test)]
    pub compiled_programs: Option<Arc<CompiledPrograms>>,
    #[cfg(not(test))]
    pub runnables: Arc<RunnableMap>,
    #[cfg(test)]
    pub runnables: Option<Arc<RunnableMap>>,
```

Reply:
Nice, most of these Options weren't necessary. address in [87c3aa2](https://github.com/FuelLabs/sway/pull/7214/commits/87c3aa2857bd46335569c7a205015a0e37494e1d)